projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -860,6 +860,14 @@ public function update_settings() {
 					$updated[ $key ] = (bool) $value;
 					break;
 
+				case 'lang_id':
+					// Due to the fact that locale variants are set in a locale_variant option, changing locale from variant to primary
+					// would look like the same lang_id is being saved and update_option would return false, even though the correct
+					// options would be set by pre_update_option_lang_id, so we should always return lang_id as updated
```

Comment:
You'll need to fix the linting error before we can merge this. I would recommend doing something like this, since it's such a long comment:

```suggestion
					/*
					 * Due to the fact that locale variants are set in a locale_variant option,
					 * changing locale from variant to primary
					 * would look like the same lang_id is being saved and update_option would return false,
					 * even though the correct options would be set by pre_update_option_lang_id,
					 * so we should always return lang_id as updated.
					 */
```